lib/vfscore/README.md

```diff
@@ -0,0 +1,131 @@
+# VFSCore: Unikraft's Virtual File System API
+
+The `vfscore` is an internal library of Unikraft that provides a generic interface of system calls related to filesystem management.
+There are multiple types of filesystems available in Unikraft (e.g: `ramfs`,` 9pfs`).
+
+## Key Functions and Data Structures
+
+The `vfscore` library provides 2 structures to define operations on a filesystem (defined in [lib/vfscore/include/vfscore/mount.h](/lib/vfscore/include/vfscore/mount.h)):
+
+```c
+struct vfsops {
+        int (*vfs_mount)        (struct mount *, const char *, int, const void*);
+        ...
+        struct vnops    *vfs_vnops;
+};
+```
+
+```c
+struct vnops {
+        vnop_open_t             vop_open;
+        vnop_close_t            vop_close;
+        vnop_read_t             vop_read;
+        vnop_write_t            vop_write;
+        vnop_seek_t             vop_seek;
+        vnop_ioctl_t            vop_ioctl;
+        ...
+};
+```
+
+The first structure mainly defines the operation of mounting the filesystem, while the second defines the operations that can be executed on files (regular files, directories, etc).
```

Comment:
I think the first structure represents the operations you can do on the filesystem, not just the mount-related ones (e.g. `sync`, `fsstat` etc.).